spec/javascripts/fixtures/json/toolbar_middleware_server_list.json

```diff
@@ -86,6 +86,10 @@
           "url_parms": "main_div"
         }
       ]
+    },
+    {
+      "id":"view_middleware",
```

Comment:
This should never affect whether a test gets run or not.

So.. the test is wrong because it's trying to go over all view toolbars without being explicit about which toolbars it should look at.

So.. if you can also fix that `Each dataView should have url set` spec please... :)

(Also, the test is completely wrong because if it's supposed to be testing that url is set, it should probably *not* be succeeding only when the url is empty.)

Reply:
>This should never affect whether a test gets run or not.
>
> So.. the test is wrong because it's trying to go over all view toolbars without being explicit about which toolbars it should look at should be there.
> 
> So.. if you can also fix that Each dataView should have url set spec please... :)

I don't follow, sorry... So it should only look at that one that I added? Do you want me to remove this change and remove the spec? Because there's no other way to get something in `$controller.dataViews`.

> (Also, the test is completely wrong because if it's supposed to be testing that url is set, it should probably not be succeeding only when the url is empty.)

`expect(dataView.url !== "").toBeTruthy();` it used to pass for everything except `""` so `expect(dataView.url).toBeTruthy();` should fail at `undefined` and `null` as well. Is that what you meant?

Reply:
The idea was, it should fail without the change the data :). (But the change is probably fine.)

(As for the second, I was confused with double negatives, sorry, you're right.. checking `dataView.url` to be truthy seems reasonable :).)

Reply:
Added check that there's at least one item, comment how to get that item and added that url is set automatically to spec name.